tubular/dates.py

```diff
@@ -663,3 +663,344 @@ def transform(self, X):
+                "winter": [12, 1, 2],  # Dec, Jan, Feb
+            }
+
+        if ("dayofweek" in include) and ("dayofweek" in self.datetime_mappings.keys()):
+            dayofweek_mapping = self.datetime_mappings["dayofweek"]
+        elif "dayofweek" in include:  # Choose default mapping
+            dayofweek_mapping = {
+                "monday": [0],
+                "tuesday": [1],
+                "wednesday": [2],
+                "thursday": [3],
+                "friday": [4],
+                "saturday": [5],
+                "sunday": [6],
+            }
+
+        # Invert dictionaries for quicker lookup
+
+        if "timeofday" in include:
+            self.timeofday_mapping = {
+                vi: k for k, v in timeofday_mapping.items() for vi in v
+            }
+            if set(self.timeofday_mapping.keys()) != set(range(24)):
+                raise ValueError(
+                    "{}: timeofday mapping dictionary should contain mapping for all hours between 0-23. {} are missing".format(
+                        self.classname(),
+                        set(range(24)) - set(self.timeofday_mapping.keys()),
+                    )
+                )
+            # Check if all hours in dictionary
+
+        if "timeofmonth" in include:
+            self.timeofmonth_mapping = {
+                vi: k for k, v in timeofmonth_mapping.items() for vi in v
+            }
+            if set(self.timeofmonth_mapping.keys()) != set(range(32)):
+                raise ValueError(
+                    "{}: timeofmonth mapping dictionary should contain mapping for all days between 1-31. {} are missing".format(
+                        self.classname(),
+                        set(range(1, 32)) - set(self.timeofmonth_mapping.keys()),
+                    )
+                )
+
+        if "timeofyear" in include:
+            self.timeofyear_mapping = {
+                vi: k for k, v in timeofyear_mapping.items() for vi in v
+            }
+            if set(self.timeofyear_mapping.keys()) != set(range(1, 13)):
+                raise ValueError(
+                    "{}: timeofyear mapping dictionary should contain mapping for all months between 1-12. {} are missing".format(
+                        self.classname(),
+                        set(range(1, 13)) - set(self.timeofyear_mapping.keys()),
+                    )
+                )
+
+        if "dayofweek" in include:
+            self.dayofweek_mapping = {
+                vi: k for k, v in dayofweek_mapping.items() for vi in v
+            }
+            if set(self.dayofweek_mapping.keys()) != set(range(7)):
+                raise ValueError(
+                    "{}: dayofweek mapping dictionary should contain mapping for all days between 0-6. {} are missing".format(
+                        self.classname(),
+                        set(range(7)) - set(self.dayofweek_mapping.keys()),
+                    )
+                )
+
+    def identify_timeofday(self, hour):
```

Comment:
very minor (and can probably be ignored) but I think conventionally these "identify functions" should be e.g. `_identify_timeofday` to signify that we don't want the user to use them

Reply:
addressed in https://github.com/lvgig/tubular/pull/49/commits/2ef53030b9b2722429d0cc56520d91f6fcf7ac89